components/scream/src/share/io/scream_output_manager.cpp

```diff
@@ -93,22 +94,30 @@ setup (const ekat::Comm& io_comm, const ekat::ParameterList& params,
   }
 
   const auto has_restart_data = (m_avg_type!=OutputAvgType::Instant && m_output_control.frequency>1);
-  if (has_restart_data) {
-    if (m_params.isSublist("Checkpoint Control")) {
-      // Output control
-      auto& pl = m_params.sublist("Checkpoint Control");
-      m_checkpoint_control.frequency  = pl.get<int>("Frequency");
-      m_checkpoint_control.frequency_units = pl.get<std::string>("Frequency Units");
-      m_checkpoint_control.nsteps_since_last_write = 0;
-
-      // File specs
-      m_checkpoint_file_specs.max_snapshots_in_file = 1;
-      m_checkpoint_file_specs.num_snapshots_in_file = 0;
-      m_checkpoint_file_specs.filename_with_time_string = pl.get("Timestamp in Filename",true);
-      m_checkpoint_file_specs.filename_with_mpiranks    = pl.get("MPI Ranks in Filename",false);
-      m_checkpoint_file_specs.filename_with_avg_type    = pl.get("AVG Type in Filename",true);
-      m_checkpoint_file_specs.filename_with_frequency   = pl.get("Frequency in Filename",true);
-    }
+  if (has_restart_data && m_params.isSublist("Checkpoint Control")) {
+    // Output control
+    // TODO: It would be great if there was an option where, if Checkpoint Control was not a sublist, we
+    //       could query the restart control information and just use that. 
+    auto& pl = m_params.sublist("Checkpoint Control");
+    m_checkpoint_control.frequency  = pl.get<int>("Frequency");
+    m_checkpoint_control.frequency_units = pl.get<std::string>("Frequency Units");
+    m_checkpoint_control.nsamples_since_last_write = 0;
+    m_checkpoint_control.timestamp_of_last_write = case_t0;
+
+    // File specs
+    m_checkpoint_file_specs.max_snapshots_in_file = 1;
+    m_checkpoint_file_specs.num_snapshots_in_file = 0;
+    m_checkpoint_file_specs.filename_with_time_string = pl.get("Timestamp in Filename",true);
+    m_checkpoint_file_specs.filename_with_mpiranks    = pl.get("MPI Ranks in Filename",false);
+    m_checkpoint_file_specs.filename_with_avg_type    = pl.get("AVG Type in Filename",true);
+    m_checkpoint_file_specs.filename_with_frequency   = pl.get("Frequency in Filename",true);
+  } else {
+    // If there is no restart data or there is but no checkpoint control sublist then we initialize
+    // the checkpoint control so that it never writes checkpoints.
+    m_checkpoint_control.frequency  = 0;
+    m_checkpoint_control.frequency_units = "never";
```

Comment:
We could use "none", to allow using CIME's `HIST_OPTION/REST_OPTION` entries directly.

Reply:
Also, if you change IOControl struct declaration in `scream_io_utils.hpp` to have
```
struct IOControl {
...
   std::string frequency_units = "none";
};
```
so that all IOControl objects are created with units="none" (hence, disabled) by default. Also, frequency is already inited to -1 in the struct decl by default, so I'm guessing setting it to 0 here is pointless.

Reply:
Good call, "never" is actually an acceptable entry for HIST/REST_OPTION, but it looks like "none" is the default from CIME so best to go with that.